apps/tron_app/tron_txn_helpers.c

```diff
@@ -0,0 +1,120 @@
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "tron_txn_helpers.h"
+
+#include <pb_decode.h>
+#include <string.h>
+#include <tron/tron.pb.h>
+
+#include "tron_priv.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+int tron_byte_array_to_raw_txn(uint8_t *byte_array,
+                               size_t size,
+                               tron_transaction_raw_t *raw_txn) {
+  if (byte_array == NULL || raw_txn == NULL)
+    return 1;
+  memzero(raw_txn, sizeof(tron_transaction_raw_t));
+
+  pb_istream_t stream = pb_istream_from_buffer(byte_array, size);
+
+  if (!pb_decode(&stream, TRON_TRANSACTION_RAW_FIELDS, raw_txn)) {
+    return -1;
+  }
+
+  return 0;
+}
+
+int tron_validate_unsigned_txn(const tron_transaction_raw_t *raw_txn) {
+  // TODO:
+  return 0;
```

Comment:
add validation like:
if contract count is non-zero
if fee limit can be non zero
if contract is not null

Reply:
added